src/main/java/gift/member/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package gift.member.service;
+
+import gift.global.exception.MemberEmailNotExistsException;
+import gift.member.dto.MemberLoginRequestDto;
+import gift.member.dto.MemberLoginResponseDto;
+import gift.member.dto.MemberRegisterRequestDto;
+import gift.member.dto.MemberResponseDto;
+import gift.member.entity.Member;
+import gift.global.exception.InvalidPasswordException;
+import gift.global.exception.MemberEmailAlreadyExistsException;
+import gift.member.repository.MemberRepository;
+import gift.global.security.JwtProvider;
+import gift.member.vo.Email;
+import gift.member.vo.Name;
+import gift.member.vo.Password;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class MemberServiceImpl implements MemberService {
+    private final MemberRepository memberRepository;
+    private final JwtProvider jwtProvider;
+
+    public MemberServiceImpl(MemberRepository memberRepository, JwtProvider jwtProvider) {
+        this.memberRepository = memberRepository;
+        this.jwtProvider = jwtProvider;
+    }
+
+    @Override
+    public void register(MemberRegisterRequestDto requestDto) {
+        if (memberRepository.findByEmail(requestDto.email()) != null) {
+            throw new MemberEmailAlreadyExistsException();
+        }
+
+        Member member = new Member(
+            null,
+            new Name(requestDto.name()),
+            new Email(requestDto.email()),
+            new Password(requestDto.password())
+        );
+        memberRepository.save(member);
+    }
+
+    @Override
+    public MemberLoginResponseDto findByEmail(MemberLoginRequestDto requestDto) {
+        Member member = memberRepository.findByEmail(requestDto.email());
+
+        if (member == null) {
+            throw new MemberEmailNotExistsException();
+        }
+
+        if (!member.getPassword().matches(requestDto.password())) {
+            throw new InvalidPasswordException();
+        }
+
+        return new MemberLoginResponseDto(jwtProvider.createToken(member));
```

Comment:
```suggestion
    @Override
    public MemberLoginResponseDto login(MemberLoginRequestDto requestDto) {
        Member member = validMember(requestDto.email(), requestDto.password());
        String token = tokenService.create(member);

        return new MemberLoginResponseDto(token);
    }

    private Member validMember(String email, String password) {
        Member member = findByEmail(email);

        check((member != null), new MemberEmailNotExistsException());
        check(member.match(password), new InvalidPasswordException());
        return member;
    }        
```
```java
public class Assert {
    public static void check(boolean condition, RuntimeException exception) {
        if (!condition) {
            throw exception;
        }
    }
}
```

- method name이 직관적이지 않은거 같아요. 기능을 잘 표현하는 이름으로 선정하는건 어떨까요?
- 반복되는 부분은 메서드 추상화를 하여 리팩토링해주세요
- 값을 꺼내서 처리하기 보다 객체에 메시지를 보내보세요.
- jwtProvider를 직접 의존하는 것보다는 토큰을 관리하는 객체를 도출하여 감싸는건 어떨까요?